### PR TITLE
explicitly numbered syllables match syllables only if numbers present

### PR DESCRIPTION
**Pull request by bcaller**
see #21 
Allows you to search text for pinyin which has been numbered.

I will modify this if I did something wrong.

### CODE REVIEW

**Comment by tsroten** [comment]
@bcaller Thanks for the pull request! I like the idea of having another constant that requires tone numbers. I'm looking it over now.

File: zhon/pinyin.py [hunk 1]
```diff
@@ -52,8 +52,11 @@
 _a_vowels = {'a': _a, 'e': _e, 'i': _i, 'o': _o, 'u': _u, 'v': _v}
 _n_vowels = {'a': 'a', 'e': 'e', 'i': 'i', 'o': 'o', 'u': 'u', 'v': 'v\u00FC'}
 
+tone_numbers_required = '[0-5]'
+tone_numbers_included = tone_numbers_required + '?'
```
Comment (tsroten) [inline zhon/pinyin.py]: Can we put an `_` in front of both of these variables so it's clear they are for internal use?

File: zhon/tests/test-pinyin.py [hunk 2]
```diff
@@ -120,6 +123,12 @@ def num_syl_to_acc(syllable):
     return syl.replace(last_vowel, _num_vowel_to_acc(last_vowel, tone))
 
 
+def random_tone(allow_toneless):
+    if allow_toneless and random.randint(1, 10) == 1:
+        return ''
+    return str(random.randint(1, 5))
```
Comment (tsroten) [inline zhon/tests/test-pinyin.py]: I like how you put this in an easy-to-call function. Is the `randint(1, 10)` (as opposed to `randint(1, 5)`) to lower the chances of a toneless syllable?

**Comment by tsroten** [comment]
This looks great. I submitted a PR to your branch with the documentation for this new feature as well. If you could merge that and add the `_` like I mentioned above, we can go ahead and get this in zhon.

**Comment by bcaller** [comment]
Just realised that you now get problems with the erhua.
I have no idea where in that monster regex I could tell it to ignore 'r2' for instance.

File: zhon/tests/test-pinyin.py (hunk 2 above)
Comment (bcaller) [inline zhon/tests/test-pinyin.py]: Yes. It's a bit silly really, so maybe I should have done something else.

**Comment by tsroten** [comment]
What problems are you having with the erhua?

**Comment by tsroten** [comment]
@bcaller I saw a comment from you about the erhua. Is there a problem with it that should prevent me from merging this PR?

**Comment by bcaller** [comment]
This regex would require a tone on the r

**Comment by bcaller** [comment]
I couldn't really decipher the regex to fix it, so I made https://github.com/bcaller/pinyin_markdown for what I wanted. Sorry. Feel free to close this, or we can think about how to fix it.

**Comment by tsroten** [comment]
@bcaller Good job on your Python Markdown extension! It looks great.

Let's leave this open for a bit while I think on it.

One easy way to deal with it is to make `xn_syl` prefix the tone numbers with something like `r?`. It's not perfect, but it's simple. Might even be worth adding a look-behind assertion to ensure the `-r` suffix isn't following an `r`.

**Pull request closed without merging**